src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Media\Storage;
+
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemInterface;
+
+class S3Storage extends FlysystemStorage
+{
+    /**
+     * @var string
+     */
+    private $endpoint;
+
+    /**
+     * @var string
+     */
+    private $bucketName;
+
+    public function __construct(FilesystemInterface $filesystem, int $segments)
+    {
+        parent::__construct($filesystem, $segments);
+
+        if (!$filesystem instanceof Filesystem || !$filesystem->getAdapter() instanceof AwsS3Adapter) {
+            throw new \RuntimeException();
```

Comment:
Wouldn't an exception text be nice? Would at the same time be a documentation for the one who reads the code. Is that something that is not supposed to happen? Also that could go into the message.